projects/ion/src/lib/picker/control-picker/control-picker.component.ts

```diff
@@ -0,0 +1,114 @@
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+
+export type TypeEvents =
+  | 'previousYear'
+  | 'previousMonth'
+  | 'nextMonth'
+  | 'nextYear'
+  | 'changeMonth'
+  | 'changeYear';
+
+type Events = {
+  type: TypeEvents;
+  value?: string;
+};
+
+export interface ControlEvent {
+  event?: Events;
+}
+
+export interface ControlPickerComponentProps {
+  month: string;
+  year: string;
+  controlPickerEvent?: EventEmitter<ControlEvent>;
+}
+
+@Component({
+  selector: 'ion-control-picker',
+  templateUrl: './control-picker.component.html',
+  styleUrls: ['./control-picker.component.scss'],
+})
+export class ControlPickerComponent {
+  @Input() month: string;
```

Comment:
```suggestion
  @Input() month: Month;
```

```
const months = [
    'Jan',
    'Fev',
    'Mar',
    'Abr',
    'Mai',
    'Jun',
    'Jul',
    'Ago',
    'Set',
    'Out',
    'Nov',
    'Dez',
  ] as const; 

type Month = typeof months[number];
```
I think that typing this variable properly can be a nice touch to your code.